tests/app/service/test_rest.py

```diff
@@ -2198,3 +2198,72 @@ def test_get_email_reply_to_addresses_with_multiple_email_addresses(client, noti
+                data=data,
+                headers=[('Content-Type', 'application/json'), create_authorization_header()])
+
+    second = json.dumps({"email_address": "second@reply.com"})
+    response = client.post('/service/{}/email-reply-to'.format(sample_service.id),
+                           data=second,
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    assert response.status_code == 201
+    json_resp = json.loads(response.get_data(as_text=True))
+    results = ServiceEmailReplyTo.query.all()
+    assert len(results) == 2
+    default = [x for x in results if x.is_default]
+    assert json_resp['data'] == default[0].serialize()
+    first_reply_to_not_default = [x for x in results if not x.is_default]
+    assert first_reply_to_not_default[0].email_address == 'first@reply.com'
+
+
+def test_add_service_reply_to_email_address_raise_exception_if_no_default(client, sample_service):
+    data = json.dumps({"email_address": "first@reply.com", "is_default": False})
+    response = client.post('/service/{}/email-reply-to'.format(sample_service.id),
+                           data=data,
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    assert response.status_code == 400
+    json_resp = json.loads(response.get_data(as_text=True))
+    assert json_resp['message'] == 'You must have at least one reply to email address as the default.'
+
+
+def test_update_service_reply_to_email_address(client, sample_service):
+    original_reply_to = create_reply_to_email(service=sample_service, email_address="some@email.com")
+    data = json.dumps({"email_address": "changed@reply.com"})
+    response = client.post('/service/{}/email-reply-to/{}'.format(sample_service.id, original_reply_to.id),
+                           data=data,
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+
+    assert response.status_code == 200
+    json_resp = json.loads(response.get_data(as_text=True))
+    results = ServiceEmailReplyTo.query.all()
+    assert len(results) == 1
+    assert json_resp['data'] == results[0].serialize()
+
+
+def test_update_service_reply_to_email_address_returns_400_when_no_default(client, sample_service):
+    original_reply_to = create_reply_to_email(service=sample_service, email_address="some@email.com")
+    data = json.dumps({"email_address": "changed@reply.com", "is_default": False})
+    response = client.post('/service/{}/email-reply-to/{}'.format(sample_service.id, original_reply_to.id),
+                           data=data,
+                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+
+    assert response.status_code == 400
+    json_resp = json.loads(response.get_data(as_text=True))
+    assert json_resp['message'] == 'You must have at least one reply to email address as the default.'
```

Comment:
can we add a test into this file for what `test_add_service_400s_when_invalid_service_id` cos I think it'll go through all the process until you insert the row, which will throw an integrity error cos of the fkey constraint - good to make sure that gets handled properly

Reply:
or maybe `404`?